src/docfx/lib/git/CommitBuildTimeProvider.cs

```diff
@@ -31,7 +32,11 @@ public CommitBuildTimeProvider(Repository repo)
 
         public bool TryGetCommitBuildTime(string commitId, out DateTime time)
         {
-            return _buildTimeByCommit.TryGetValue(commitId, out time);
+            if (!_buildTimeByCommit.TryGetValue(commitId, out time))
```

Comment:
can we use GetOrAdd here, then `UpdateAndSaveCache` don't need retrieve commits again?

Reply:
`UpdateAndSaveCache` have to set the time for commits that are not retrieved using this method, so GetOrAdd here does not solve the problem.